From 636b598dc427dcccd3b09079cb14b36dd09eebcb Mon Sep 17 00:00:00 2001
From: Murala S Swamy <ssmurala@habana.ai>
Date: Tue, 05 Nov 2024 14:20:35 +0200
Subject: [PATCH] [FSW-8694] fs1: switch to control slave spi fuse

Fake fuse boot pointer doesnt use key index
because of no security key usage in fake fuse.
Hence, key index flag will be used to control
slave fake fuse initialization. If it is set
255, master will not write slave fake fuse
values on slave scratchpad register.

Change-Id: I9d8c9167037da7413ee1f9590a22d889cf78d0ad
---

diff --git a/board/habanalabs/fs1_arc/post-image-spi.sh b/board/habanalabs/fs1_arc/post-image-spi.sh
index 40077f0..d98e307 100755
--- a/board/habanalabs/fs1_arc/post-image-spi.sh
+++ b/board/habanalabs/fs1_arc/post-image-spi.sh
@@ -532,6 +532,11 @@
 	declare -A SPI_REGION_OFFSETS
 	declare -A SPI_REGION_SIZES
 
+	if [ ${IMG_TYPE} == ${IMAGE_TYPE_FAKE_FUSE} ]
+	then
+		KEY_NUM=0 # 255 to stop, < 255 to write slave Fake fuse values.
+	fi
+
 	hl_parse_spi_flash_regions SPI_REGION_OFFSETS SPI_REGION_SIZES fs1
 
 	# Prepare empty space for size of boot pointer and size of image pointer.
